### ISSUE TITLE
Theme editor: can't set background-image on Global button

### ISSUE THREAD

**Issue by JonFerraiolo** [maqetta/maqetta#2818]
@billreed63 @childsb 

I cloned ipad theme, then tried to put an image as the background for the global button widget. While I was able to set gradients or colors, choosing an image shows the correct preview in the Background dialog, but clicking OK causes no effect on the canvas of the theme editor.

**Comment by JonFerraiolo** [comment]
I debugged into this issue. I think I figured out what's wrong. We are doing "URL path math" somewhere in our code to compute the relative URL that will point to the image file. 

The "path math" is figuring out the relative URL for the image file using the base theme CSS file as the base path, which means something like `lib/dojo/dijit/themes/android/android.css` is the presumed base path. As a result, if you pick `samples/SampleBanner.jpg` in the Background dialog, the relative URL will have 5 `../` in it, resulting in `../../../../../samples/SampleBanner.jpg`.

However, the CSS property is going to be written into the delta theme CSS file, not the base theme CSS file. Therefore, the proper relative URL should instead only have 2 `.../` in it. In fact, when I bring up the delta theme file in the source editor and manually change the 5 `.../` to 2 `.../`, then I see the background image in the page editor.

I haven't found the code yet which does the path math.

**Comment by JonFerraiolo** [comment]
Reassigning to @billreed63 because this is centered on delta themes interacting with the theme editor.

Further study shows that the relative path math is done in BackgroundDialog.js using `this._baseLocation`, which it gets from Background.js, which gets it from this code at Cascade.js:75:

```
                    if(widget._setBaseLocationAttr){
                        widget.set('baseLocation', (loc && loc.getPath) ?loc.getPath():null); //#23 FIXME why no getPath
                    }
```

which gets called by `_onChange` which at line 1-26 computes the location from CSSFile pointed to by a rule, which was passed in by `selectRule() (Cascade.js:745)`.

So, somehow or other we need to pass in the path to the delta CSS file rather than the original/base theme's CSS file.

**Comment by billreed63** [comment]
fixed by #2422
